cclib/parser/molcasparser.py

```diff
@@ -725,6 +738,62 @@ def extract(self, inputfile, line):
                     self.ccenergies= []
                 self.ccenergies.append(ccenergies)
 
+        #  ++    Primitive basis info:
+        #        ---------------------
+        #
+        #
+        #                      *****************************************************
+        #                      ******** Primitive Basis Functions (Valence) ********
+        #                      *****************************************************
+        #
+        #
+        #   Basis set:C.AUG-CC-PVQZ.........                                                          
+        #
+        #                    Type         
+        #                     s
+        #             No.      Exponent    Contraction Coefficients
+        #             1  0.339800000D+05   0.000091  -0.000019   0.000000   0.000000   0.000000   0.000000
+        #             2  0.508900000D+04   0.000704  -0.000151   0.000000   0.000000   0.000000   0.000000
+        # ...
+        # ...
+        #             29  0.424000000D+00   0.000000   1.000000
+        #
+        #   Number of primitives                                   93
+        #   Number of basis functions                              80
+        #
+        #  --
+        if '++    Primitive basis info:' in line:
+            self.skip_lines(inputfile, ['d', 'b', 'b', 's', 'header', 's', 'b'])
+            line = next(inputfile)
+            self.gbasis_array = []
+            while '--' not in line and '****' not in line:
+                if 'Basis set:' in line:
+                    basis_element = line.split()[1].split('.')[0].split(':')[1]
+                    basis_element = basis_element[0] + basis_element[1:].lower()
+                    self.gbasis_array.append((basis_element, []))
+
+                if 'Type' in line:
+                    exponents = []
+                    coefficients = []
+                    line = next(inputfile)
+                    _type = line.split()[0].upper()
+                    func_array = []
+                    self.skip_line(inputfile, 'headers')
+                    line = next(inputfile)
+                    while line.split():
+                        exponents.append(self.float(line.split()[1]))
+                        coefficients.append([self.float(i) for i in line.split()[2:]])
+                        line = next(inputfile)
+
+                    for i in range(len(coefficients[0])):
+                        func_tuple = (_type, [])
+                        for j in range(len(exponents)):
+                            if coefficients[j][i] != 0:
```

Comment:
I think you also need to make sure there isn't `1.0` in any other column. The output for the first set of S basis functions have 3 extra values compared with Q-Chem.

Reply:
I think the outputs from other parsers also differ a little bit from each other. Let me look up a few examples and get back to you on this.

Reply:
I think we can get to the root of this in a separate effort. If not, @ATenderholt, let's continue the conversation in #632.